server/pbench/bin/index-pbench

```diff
@@ -321,6 +368,115 @@ class ResultData(object):
             datafiles.append(datafile)
         return datafiles
 
+    @staticmethod
+    def expand_template(templ, d):
+        return templ
+
+    @staticmethod
+    def make_datum_wrapper(templ):
+        # make a shallow copy
+        d = templ.copy()
+        del d['timeseries']
+        # construct the uid from the template and the values in d
+        # and replace the template with the result
+        d['uid'] = ResultData.expand_template(d['uid'], d)
+        return d
+
+    @staticmethod
+    def gen_sample_sources(results, benchmark):
+
+        # sample results is a dict with either two or three (top-level) entries:
+        # latency results have three entries: parameters.benchmark, latency.usec, throughput.trans_sec.
+        # throughput results have two entries: parameters.benchmark, throughput.Gb_sec.
+        # All entries are lists, but paremeters.benchmark is a one-element list, whereas the other lists
+        # may have multiple elements, one for a cumulative entry ("all" clients/servers) and one for each
+        # pair of client/server in the experiment (at least for uperf - fio is TBD).
+
+        if 'latency' in results:
+            # Latency results include a latency['usec'] list (or
+            # latency['clat'|'slat'|???] in the fio case) *and* a
+            # throughput['trans_sec'] list (or throughput['iops_sec']
+            # in the fio case). Each list has elements for "all" and
+            # for each client/server pair (or client?) used in the test.
+            if benchmark == "uperf":
+                latency_sublist = ['usec']
+            elif benchmark == "fio":
+                latency_sublist = ['clat', 'lat', 'slat']
+            else:
+                return
+            for sublist in latency_sublist:
+                latency_list = results['latency'][sublist]
+                for i in range(len(latency_list)):
```

Comment:
Can we consider using `for idx,latency_el in enumerate(latency_list):` instead of `range(len())`?

Reply:
Sure.